### PR TITLE
ensure `gr.Dataframe` updates as expected

### LINKED ISSUE #6252: Gradio.Dataframe does no longer properly pass values updated in the GUI when used as input from version 4
### Describe the bug

When running the [filter records example](https://www.gradio.app/docs/dataframe) code in `3.50.2` the records are filtered based on gender-entries a user made in the webpage and based on the selected gender in the dropdown. These records are displayed in the output dataframe as expected.

However, in gradio version `4.0.2` the values that are entered in the webpage are not properly passed to the backend. Thus, no output is visible in the output dataframe.

Changing the `filter_records(..)` to include a print-statement shows that the values from the GUI are not reaching the backend:
```
def filter_records(records, gender):
    print(records)
    return records[records["gender"] == gender]
```

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

1. Install gradio version `4.0.2`
2. Add the code from the [filter records example](https://www.gradio.app/docs/dataframe) to a file `df_example.py`
3. Run `python df_example.py`
4. Optionally add the print-statement as mentioned before
5. Enter values in the interface and press `Submit`
6. Observe that the output dataframe stays empty

### Screenshot

_No response_

### Logs

```shell
No (error) logs given
```

### System Info

```shell
gradio environment
Gradio Environment Information:
------------------------------
Operating System: Windows
gradio version: 4.0.2
gradio_client version: 0.7.0

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
altair: 5.1.2
fastapi: 0.104.1
ffmpy: 0.3.1
gradio-client==0.7.0 is not installed.
httpx: 0.25.0
huggingface-hub: 0.18.0
importlib-resources: 6.1.0
jinja2: 3.1.2
markupsafe: 2.1.1
matplotlib: 3.7.0
numpy: 1.23.5
orjson: 3.9.10
packaging: 22.0
pandas: 1.5.3
pillow: 9.4.0
pydantic: 2.4.2
pydub: 0.25.1
python-multipart: 0.0.6
pyyaml: 6.0
requests: 2.28.1
semantic-version: 2.10.0
tomlkit==0.12.0 is not installed.
typer: 0.9.0
typing-extensions: 4.8.0
uvicorn: 0.23.2
websockets: 11.0.3
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2023.10.0
httpx: 0.25.0
huggingface-hub: 0.18.0
packaging: 22.0
requests: 2.28.1
typing-extensions: 4.8.0
websockets: 11.0.3
```

### Severity

I can work around it

### PR DESCRIPTION
**Pull request by pngwn**
## Description

Closes #6252.

Silly omissions when refactoring. Also fixed a weird bug that didn't really make sense to me but seems to work ok now.

Closes: #(issue)

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by pngwn** [comment]
unit test would have caught this, i don't have time to write one now but i'll add it to the list.

File: js/dataframe/Index.svelte [hunk 1]
```diff
@@ -99,6 +99,7 @@
 		{col_count}
 		{value}
 		{headers}
+		on:change={(e) => (value = e.detail)}
```
Comment (abidlabs) [inline js/dataframe/Index.svelte:102]: Ah!!!

Comment (pngwn) [reply js/dataframe/Index.svelte:102]: yeee

Comment (pngwn) [reply js/dataframe/Index.svelte:102]: very upsetting that it took me so long to find it tbh. i guess you often miss the obvious stuff.

**Comment by abidlabs** [comment]
Wait hmm I'm not sure if this fixed it for me

**Comment by pngwn** [comment]
hmmmm

**Comment by abidlabs** [comment]
The issue now is that if I change the value of a cell and blur, it reverts to the original value

**Comment by pngwn** [comment]
well

**Comment by pngwn** [comment]
i wonder if i broke that with my fix

File: js/dataframe/Index.svelte (hunk 1 above)
Comment (abidlabs) [reply js/dataframe/Index.svelte:102]: I spent a couple hours trying different things so I feel like you got it pretty fast 😄

**Comment by abidlabs** [comment]
Well in one sense the original bug is now fixed since you can't change the value of the dataframe at all 😆

**Comment by abidlabs** [comment]
But yeah I think the blur changes are breaking something

**Comment by pngwn** [comment]
yeh, i broke it sec.

**Comment by pngwn** [comment]
okay that should do it, there is a small bug around clicking but i think we should get this out now and fix the click bug later. It isn't too awful.

**Comment by pngwn** [comment]
File an issue: https://github.com/gradio-app/gradio/issues/6291

**Comment by abidlabs** [comment]
LGTM @pngwn!

**Pull request merged**